### PR TITLE
Global fetch method

### LINKED ISSUE #216: Frontend: Global fetch method
## **User story** 
**Who:** As a developer
**What:** I want to reuse a fetch method to comminucate with the backend api. 
**Why:**  To not have to implement the same logic in every component and to ensure that the user authentication token is atteched in the header

Example:
```
 const { getUserToken } = useFirebase();
 const token = await getUserToken();
  const response = await fetch('/api/users', {
    method: 'POST',
    mode: 'cors',
    headers: {
      'Content-type': 'application/json',
      authorization: `Bearer ${token}`,
    },
    body: JSON.stringify(something),
  });
```
**Acceptance criteria:**

- The method must be flexible enough to work under all circumstances. 
- It must have at least the url and a method verb as parameters.
- It must send the authentication token in the header
- The function must throw if the ok property is not true. The error thrown must include the error message and the status code

### PR DESCRIPTION
**Pull request by NatsGt**
# Description

Fetch method that will work with GET, POST, PUT and DELETE

Fixes #216 

# How to test?

Set the environment on production (NODE_ENV=production) and test the fetch with the following parameters: 

1. endpoint (without /api/users/)
2. method
3. object to be passed to the body if is POST or PUT

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have followed the name conventions for CSS Classnames and filenames, Components names and filenames, Style filenames, if you are in doubt check the the project README.MD and here https://github.com/HackYourFuture-CPH/curriculum/blob/master/review/review-checklist.md
- [x] I have commented my code, particularly in hard-to-understand areas, if you code was simple enough mark the box anyway
- [x] I have made corresponding changes to the documentation, if you code was simple enough mark the box anyway
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/client/containers/fetchMethod/fetch.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { getUserToken } from '../../firebase/googleAuth';
+import { ApiError } from '../../ErrorBoundary';
+
+export async function fetchFromDb(endpoint, fetchMethod, postBody = {}) {
+  const token = await getUserToken();
```
Comment (Midnighter) [inline src/client/containers/fetchMethod/fetch.js]: In case the token needs to be refreshed and the request fails... does this need a `try` & `catch`?

**Comment by Midnighter** [comment]
Sometimes the line annotations are bad, I meant a `try` `catch` around

```js
const token = await getUserToken();
```

**Review by Midnighter — APPROVED** [review]
That looks good. Now to use this function everywhere... 😜

**Review by senner007 — APPROVED** [review]
✔

**Pull request merged**